rewrite-java-17/src/main/java/org/openrewrite/java/isolated/ReloadableJava17ParserVisitor.java

```diff
@@ -342,19 +342,25 @@ public J visitBreak(BreakTree node, Space fmt) {
     @Override
     public J visitCase(CaseTree node, Space fmt) {
         J.Case.Type type = node.getCaseKind() == CaseTree.CaseKind.RULE ? J.Case.Type.Rule : J.Case.Type.Statement;
+
+        JContainer<J> caselabels = JContainer.build(
+                node.getLabels().isEmpty() ? EMPTY : sourceBefore("case"),
+                node.getLabels().isEmpty() || node.getLabels().get(0) instanceof DefaultCaseLabelTree ?
+                        List.of(JRightPadded.build(new J.Identifier(randomId(), Space.EMPTY, Markers.EMPTY, emptyList(), skip("default"), null, null))) :
+                        convertAll(node.getLabels(), commaDelim, t -> EMPTY),
+                Markers.EMPTY
+        );
+        JContainer<Expression> expressions = JContainer.build(caselabels.getBefore(), caselabels.getElements().stream().filter(Expression.class::isInstance).map(Expression.class::cast).map(JRightPadded::build).toList(), caselabels.getMarkers());
+
         return new J.Case(
                 randomId(),
                 fmt,
                 Markers.EMPTY,
                 type,
                 null,
-                JContainer.build(
-                        node.getExpressions().isEmpty() ? EMPTY : sourceBefore("case"),
-                        node.getExpressions().isEmpty() ?
-                                List.of(JRightPadded.build(new J.Identifier(randomId(), Space.EMPTY, Markers.EMPTY, emptyList(), skip("default"), null, null))) :
-                                convertAll(node.getExpressions(), commaDelim, t -> EMPTY),
-                        Markers.EMPTY
-                ),
+                expressions,
+                caselabels,
+                null,
```

Comment:
Why don't we have a `convert(node.getGuard())` here?

Reply:
Because Java 17 doesn't have a guard

Reply:
Makes sense! I thought I saw a `getGuard()` method, but I must have been looking at the Java 21 version of `CaseTree`.